pkg/trace/config/config.go

```diff
@@ -170,9 +167,6 @@ func (c *AgentConfig) APIKey() string {
 
 // Validate validates if the current configuration is good for the agent to start with.
 func (c *AgentConfig) validate() error {
-	if len(c.Endpoints) == 0 || c.Endpoints[0].APIKey == "" {
```

Comment:
We shouldn't move this. I want this function to be the ultimate truth on whether a Config is ready to be used by the trace-agent.